src/NamedTupleUtilities.jl

```diff
@@ -51,13 +51,44 @@ julia> range((a=1,b=2,c=3),Val(:a),Val(:b))
             push!(names, n)
         end
         if n == cn
+            # rangeStarted = false
+            break
+        end
+    end
+    types = Tuple{(fieldtype(a, n) for n in names)...}
+    vals = Expr[:(getfield(a, $(QuoteNode(n)))) for n in names]
+    return :( NamedTuple{$(names...,),$types}(($(vals...),)) )
+end
+
+"""
+    range(a::NamedTuple, b::Val{n}, c::Val{n})
+Return a NamedTuple which retains `b`th to `c`th fields in `a`. 
+If `b` is greater than or equal to `c`, then it will return the empty NamedTuple.
+```jldoctest
+julia> QueryOperators.NamedTupleUtilities.range((a=1,b=2,c=3),1,2)
+(a = 1, b = 2)
+```
+"""
+@generated function range(a::NamedTuple{an}, b::Int, c::Int) where {an}
```

Comment:
I think this needs to be ``@generated function range(a::NamedTuple{an}, ::Val{b}, ::Val{c}) where {an, b:<Int, c<:Int}``? I don't think the current version can work, see my next comment.